admin/class-insert-rule-data.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Inserts rule data about a post to the database
+ *
+ * @since 2.0.0
+ *
+ * @package Accessibility_Checker
+ */
+
+namespace EDAC\Admin;
+
+/**
+ * Class for inserting rule data into the database
+ *
+ * @since 2.0.0
+ */
+class Insert_Rule_Data {
+
+	/**
+	 * Insert rule data into database
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param object $post     The post object.
+	 * @param string $rule     The rule.
+	 * @param string $ruletype The rule type.
+	 * @param string $rule_obj The object.
+	 *
+	 * @return void|int
+	 */
+	public function insert( object $post, string $rule, string $ruletype, string $rule_obj ) {
+
+		if ( ! $post || ! $rule || ! $ruletype || ! $rule_obj ) {
```

Comment:
Since these are typed parameters these checks can be more robust. `empty()` checks for `0`, `null`, `false`, an empty string `''`, and an unset variable, covering more cases where the variable might not be useful.

Example:
```php
if ( empty( $post->ID ) || empty( $rule ) || empty( $ruletype ) || empty( $rule_obj ) ) {
    return;
}
```